### DISCUSSION TITLE
Display python symbol version and virtualenv name only if virutalenv is activated

### DISCUSSION THREAD

**Discussion by matthaeusheer** [starship/starship#1906] [Q&A]
The title says it all. How would I achieve that?

So far I use
```
[python]
symbol = " "
format = "[\u2571 ](bg:237)[$symbol$version $virtualenv](bg:237 $style)"
```
which makes the python symbol and version pop up when I enter say a directory in which a Pipfile is present. However I want those information only to be displayed, when I actually activated the environment via `pipenv shell`.

Cheers,
Matt

**Comment by badnetmask** [comment]
[According to the documentation](https://starship.rs/config/#python), all you need to do is set `scan_for_pyfiles=false`.

**Reply by matthaeusheer** [reply]
Cheers for pointing that out. I tried both scan_for_pyfiles and pyenv_version_name, but same behaviour:
```
[python]
symbol = " "
pyenv_version_name = false
scan_for_pyfiles = false
format = "[\u2571 ](bg:237)[${symbol}${version} ${virtualenv}](bg:237 $style)"
```
The python version is being displayed even though no virtualenv is activated. It does switch however to the proper virtualenv version incl. the virtualenv name in paranthesis when is is activated.

**Reply by davidkna** [reply]
`scan_for_pyfiles` prevents looking for files with the `*.py` extension but `requirements.txt` `.python-version` `pyproject.toml` `Pipfile` `tox.ini` `setup.py` and `__init__.py` still trigger the module.

**Comment by andytom** [comment]
Hi @matthaeusheer, this isn't possible at the moment, only displaying the python module if in a virtualenv, if you open a feature request we can take a look at this.

**Comment by pure-chance** [comment]
For the record, I had the same idea and I've finally found the solution. This is my python section

```toml
[python]
format = '[(\($virtualenv\)) ](black)'
```

what this does is show ($virtualenv) if $virtualenv exists and is activated, and shows nothing otherwise. The trick is using conditional format strings. The outer () means that $virtualenv will only appear if the variable is not none, which in the case of virtualenv's is when the virtualenv is not activated. The inner \(\) are the paranthesis that are displayed to the screen. Conditional format strings are explained in the configuration section of the website, so you can read up on how it works in more detail.

**Comment by Klaudia-Schulze** [comment]
I'm using below config. This may seem unusual but it works. and only shows python version only when virtual env is activated.
```python
[python]
symbol = ""
detect_extensions = []
detect_files = []
style = "bg:color12"
format = '[[ $symbol( $version) ](fg:color13 bg:color12)]($style)'
```
